frontend/src/pages/notebookController/NotebookController.tsx

```diff
@@ -1,12 +1,18 @@
 import * as React from 'react';
+import { Navigate } from 'react-router-dom';
 import { useUser } from '~/redux/selectors';
+import { useCheckJupyterEnabled } from '~/utilities/notebookControllerUtils';
 import NotebookServerRoutes from './screens/server/NotebookServerRoutes';
 import NotebookControllerTabs from './screens/admin/NotebookControllerTabs';
 import { NotebookControllerContextProvider } from './NotebookControllerContext';
 import ValidateNotebookNamespace from './ValidateNotebookNamespace';
 
 const NotebookController: React.FC = () => {
   const { isAdmin } = useUser();
+  const isJupyterEnabled = useCheckJupyterEnabled();
+  if (!isJupyterEnabled) {
```

Comment:
Is there any other way to pull this workaround? I'm not agains it, I mean we would need to deprecate this anyway, but still. either way we can keep it.

Reply:
We need an additional check that is async to conditionally enable / disable extension contributions. But we don't have anything other than feature flags / areas at the moment.
This can be added in the future but for now this seems like a simple work around.